portal-bridge/src/beacon_bridge.rs

```diff
@@ -78,17 +75,20 @@ impl BeaconBridge {
     ///  Get and serve the latest beacon data.
     async fn launch_latest(&self) {
         // Current sync committee period known by the bridge
-        let current_period = Arc::new(AtomicUsize::new(0));
+        let mut current_period: usize = 0;
 
         loop {
             let consensus_api = self.api.clone();
             let portal_clients = self.portal_clients.clone();
-            let current_period = Arc::clone(&current_period);
 
-            tokio::spawn(async move {
-                Self::serve_latest(consensus_api, portal_clients, current_period).await;
+            let latest_result = tokio::spawn(async move {
+                Self::serve_latest(consensus_api, portal_clients, current_period).await
             });
 
+            current_period = latest_result
+                .await
+                .expect("Serving latest data task should always succeed.");
```

Comment:
I'm not sure if this is a valid assumption. I guess we still want to request the data for the next slot even if the request for the current slot fails for some reason.

Reply:
`serve_latest()` is infallible. This await will only fail if the task gets cancelled by tokio, which we're not doing explicitly anywhere. Since cancelling this task should only happen on shutdown, we shouldn't ever trigger this panic.

Reply:
You are right, this should not panic :yum:.